src/gui_common/CustomPopup.cs

```diff
@@ -0,0 +1,132 @@
+﻿using Godot;
+
+/// <summary>
+///   A custom popup that gives us more control over <see cref="Popup"/>.
```

Comment:
I suppose CustomWindow would be a good alternative (going by Godot 4's terminology)?

Popup, dialogs, closable panels are in a way a window. I used Popup here as an umbrella term for controls that are top level.

Reply:
I think that would be a worthwhile separation to create with the naming. Btw I just remembered that we have some stuff regarding the popups in the styleguide so that document should also be updated.

Reply:
Done, thanks for reminding.